app/custom-oauth/server/custom_oauth_server.js

```diff
@@ -248,6 +250,8 @@ export class CustomOAuth {
 				if (!identity.email && (identity.emails && Array.isArray(identity.emails) && identity.emails.length >= 1)) {
 					identity.email = identity.emails[0].address ? identity.emails[0].address : undefined;
 				}
+				const user = Users.findOneByEmailAddress(identity.email);
+				mapRolesFromSSO(user, identity, this.rolesClaim || 'roles');
```

Comment:
Do we have a setting for mapping roles?  If so we should use it.  Especially if an oauth provider doesn’t have roles... it would strip the user and admin roles from them

Reply:
not yet, but i can add an option for that

Reply:
did i need to also add a translation again for all locales?

Reply:
No definitely not, we usually just require the key is added and that's it.  With the other key you went above and beyond :)